presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java

```diff
@@ -137,7 +139,8 @@ public StoragePartitionLoader(
         this.hdfsContext = new HdfsContext(session, table.getDatabaseName(), table.getTableName(), table.getStorage().getLocation(), false);
         this.fileIterators = requireNonNull(fileIterators, "fileIterators is null");
         this.schedulerUsesHostAddresses = schedulerUsesHostAddresses;
-        this.hoodiePathFilterSupplier = Suppliers.memoize(HoodieROTablePathFilter::new)::get;
+        this.hoodiePathFilterLoadingCache = CacheBuilder.newBuilder().build(CacheLoader.from(
```

Comment:
Is it possible for this cache to grow unbounded?

Reply:
Ideally it won't but as a preventive measure I will make a change to limit the maximum size of the cache.